src/index.ts

```diff
@@ -0,0 +1,316 @@
+/**
+ * @fileOverview Exports the Sajari javascript api.
+ * @name index.ts
+ * @author Sajari
+ * @license MIT
+ * @module sajari
+ */
+
+import {
+  newRequest,
+  newAggregates,
+  newQueryID,
+  newResult,
+  newResults,
+  valueFromProto
+} from "./constructors";
+import { newError, SearchError } from "./error";
+import { AggregateResponse, Result, Results, ResultValues } from "./results";
+
+const UserAgent = "sdk-js-1.0.0";
+
+export interface Tracking {
+  // Tracking specifies which kind (if any) tokens should be generated and returned
+  // with the query results.
+  type: TrackingType;
+
+  // QueryID is a unique identifier for a single search query.  In the
+  // case of live querying this is defined to be multiple individual queries
+  // (i.e. as a user types the query is re-run).
+  query_id: string;
+
+  // Sequence (i.e. sequential identifier) of this  in the context of a
+  // sequence of queries.
+  sequence: number;
+
+  // Field is the field to be used for adding identifier information to
```

Comment:
This isn't clear.  Needs to be a field on each returned result, and must uniquely define the result.